modules/shared/src/main/scala/org/tessellation/ext/fs2.scala

```diff
@@ -54,4 +58,12 @@ object fs2 {
     }
   }
 
+  implicit class StreamOps[F[_]: Sync, A](s: Stream[F, A]) {
+
+    def evalMapLocked[B](semaphores: NonEmptyList[Semaphore[F]])(fn: A => F[B]): Stream[F, B] =
```

Comment:
I'd write there a second fn:
```scala
def evalMapLocked[B](semaphore: Semaphore[F])(fn: A => F[B]): Stream[F, B] =
  evalMapLocked(NonEmptyList.one(semaphore))(fn)
```
because most of the time you pass one semaphore only

Reply:
Why not!